azkaban-common/src/main/java/azkaban/executor/ExecutionOptions.java

```diff
@@ -176,27 +180,39 @@ public static ExecutionOptions createFromObject(final Object obj) {
    * Validate the ExecutionOption's Flow-Parameters against the application level Properties
    * @throws ServletException if any of the parameter is invalid
    */
-  public void validateFlowParameters(Props azProps)
+  public void validatePreprocessFlowParameters(Props azProps)
       throws ServletException {
     List<String> errMsg = new ArrayList<>();
 
     Map<String, String> flowParameters = this.getFlowParameters();
     if (flowParameters == null || flowParameters.isEmpty()) {
       return;
     }
+    // if ALLOW_RESTART_ON_EXECUTION_STOPPED then ALLOW_RESTART_ON_STATUS += "EXECUTION_STOPPED,"
+    if (Boolean.parseBoolean(
+        flowParameters.getOrDefault(
+            FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_EXECUTION_STOPPED, "false"))
+    ){
+      flowParameters.put(
+          FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS,
+          Status.EXECUTION_STOPPED.name() + "," +
+              flowParameters.getOrDefault(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS, "")
+      );
+    }
     if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS)) {
-
-      // user defined list
-      final String[] statuses = flowParameters
+      // user defined list; remove empty spaces
+      final List<String> statuses = Arrays.stream(flowParameters
           .getOrDefault(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS, "")
-          .split("\\s*,\\s*");
+          .split("\\s*,\\s*")).filter(s -> !s.trim().isEmpty()).collect(Collectors.toList());
```

Comment:
Could you please check if this line compiles? There seem to be an extra closing `)` after split.

Reply:
Oh it's because adding `Arrays.stream()` to convert to stream type